src/modules/cpu/kernel/resize_crop_mirror.hpp

```diff
@@ -0,0 +1,1012 @@
+        __m256 pWRatio = _mm256_set1_ps(wRatio);
+        __m256 pWOffset = _mm256_set1_ps(wOffset);
+        __m256i pxMaxSrcLoc = _mm256_set1_epi32(maxWidthLimitMinusStride);
+        __m256 pWeightParams[INTERP_BILINEAR_NUM_COEFFS], pBilinearCoeffs[INTERP_BILINEAR_NUM_COEFFS], pDstLoc;
+        Rpp32f weightParams[INTERP_BILINEAR_NUM_COEFFS], bilinearCoeffs[INTERP_BILINEAR_NUM_COEFFS];
+        Rpp32s srcLocationColumnArray[8] = {0};     // Since 8 dst pixels are processed per iteration
+        Rpp32s srcLocationRow, srcLocationColumn;
+        Rpp32u mirrorFlag = mirrorTensor[batchCount];
+        Rpp32u width = dstImgSize[batchCount].width;
+
+        __m256 pDstLocInit =  avx_pDstLocInit;
+        auto computeFnSrcLocAvx = &compute_resize_bilinear_src_loc_and_weights_avx;
+        if(mirrorFlag)
+        {
+            pDstLocInit =  _mm256_setr_ps(width - 1, width - 2, width - 3, width - 4, width - 5, width - 6, width - 7, width - 8);
+            computeFnSrcLocAvx = &compute_resize_bilinear_src_loc_and_weights_mirror_avx;
+        }
+
+        // Resize Crop Mirror with fused output-layout toggle (NHWC -> NCHW)
+        if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NCHW))
+        {
+            Rpp8u *dstPtrRowR, *dstPtrRowG, *dstPtrRowB;
+            dstPtrRowR = dstPtrChannel;
+            dstPtrRowG = dstPtrRowR + dstDescPtr->strides.cStride;
+            dstPtrRowB = dstPtrRowG + dstDescPtr->strides.cStride;
+
+            for(int dstLocRow = 0; dstLocRow < dstImgSize[batchCount].height; dstLocRow++)
+            {
+                Rpp8u *dstPtrTempR, *dstPtrTempG, *dstPtrTempB;
+                dstPtrTempR = dstPtrRowR;
+                dstPtrTempG = dstPtrRowG;
+                dstPtrTempB = dstPtrRowB;
+
+                Rpp8u *srcRowPtrsForInterp[2];     // kernelSize(2)
+                compute_resize_bilinear_src_loc_and_weights(dstLocRow, hRatio, srcLocationRow, &weightParams[0], hOffset); // Compute the src row location correspoding to the dst row location
+                compute_src_row_ptrs_for_bilinear_interpolation(srcRowPtrsForInterp, srcPtrChannel, srcLocationRow, maxHeightLimit, srcDescPtr); // Compute the src row pointers for interpolation
+                pWeightParams[0] = _mm256_set1_ps(weightParams[0]);
+                pWeightParams[1]  = _mm256_set1_ps(weightParams[1]);
+                pDstLoc = pDstLocInit;
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrementPerChannel)
+                {
+                    __m256 pSrc[12], pDst[3];
+                    __m256i pxSrcLoc;
+                    computeFnSrcLocAvx(pDstLoc, pWRatio, srcLocationColumnArray, &pWeightParams[2], pxSrcLoc, pWOffset, true);   // Compute the src col location correspoding to the dst col location
```

Comment:
Is pDstLoc changing with every call of the helper fn? Else use pDstLocInit instead

Reply:
Yeah, pDstLoc changes inside the helper, so an initial value has been set as pDstLocInit, and pDstLoc is used inside - https://github.com/r-abishek/rpp/blob/ar/opt_resize_crop_mirror/src/include/cpu/rpp_cpu_common.hpp#L4470